addons/crm_mail_client_extension/controllers/main.py

```diff
@@ -0,0 +1,108 @@
+        return werkzeug.utils.redirect('/web#action=%s&model=crm.lead&partner_id=%s' % (server_action.id, int(partner_id)))
+
+    @http.route('/mail_client_extension/lead/create', type='json', auth='outlook', cors="*")
+    def crm_lead_create(self, partner_id, email_body):
+        partner = request.env['res.partner'].browse(partner_id).exists()
+        if not partner:
+            return {'error': 'partner_not_found'}
+
+        record = request.env['crm.lead'].create({
+            'name': _("%s's opportunity", partner.name),
+            'partner_id': partner_id,
+            'description': html2plaintext(email_body),
+        })
+
+        return {'lead_id': record.id}
+
+    def _fetch_partner_leads(self, partner, limit=5, offset=0):
+        """
+        Returns an array containing partner leads, each lead will have the following structure :
+        {
+            id: the lead's id,
+            name: the lead's name,
+            expected_revenue: the expected revenue field value
+            probability: the value of the probability field,
+            recurring_revenue: the value of the recurring_revenue field if the lead has a recurring revenue
+            recurring_plan: the value of the recurring plan field if the lead has a recurring revenue
+        }
+        """
+
+        partner_leads = request.env['crm.lead'].search(
+            [('partner_id', '=', partner.id)], offset=offset, limit=limit)
+        recurring_revenues = request.env.user.has_group('crm.group_use_recurring_revenues')
+
+        leads = []
+        for lead in partner_leads:
+            lead_values = {
+                'id': lead.id,
+                'name': lead.name,
+                'expected_revenue': formatLang(request.env, lead.expected_revenue, monetary=True,
+                                               currency_obj=lead.company_currency),
+                'probability': lead.probability,
+            }
+
+            if recurring_revenues:
+                lead_values.update({
+                    'recurring_revenue': formatLang(request.env, lead.recurring_revenue, monetary=True,
+                                                    currency_obj=lead.company_currency),
+                    'recurring_plan': lead.recurring_plan.name,
+                })
+
+            leads.append(lead_values)
+
+        return leads
+
+    def _prepare_contact_values(self, partner):
+        contact_values = super(MailClientExtensionController, self)._prepare_contact_values(partner)
+        if not partner:
+            contact_values['leads'] = []
+        else:
+            contact_values['leads'] = self._fetch_partner_leads(partner)
+        return contact_values
+
+    def _mail_content_logging_models_whitelist(self):
+        loggable_models = super(MailClientExtensionController, self)._mail_content_logging_models_whitelist()
+        loggable_models.append('crm.lead')
```

Comment:
You can do
`return super(MailClientExtensionController, self)._mail_content_logging_models_whitelist() + ['crm.lead']`

To be a bit more brief.